cosmo_tester/test_suites/stress_test_openstack/many_deployments_to_single_manager_parallel_test.py

```diff
@@ -0,0 +1,250 @@
+        manager_disk_space_total = self.get_manager_disk_total()
+        manager_memory_total = self.get_manager_memory_total()
+        self.state_dict = {"static_info":
+                           {"manager_disk_space_total":
+                            manager_disk_space_total,
+                            "manager_memory_total":
+                            manager_memory_total}}
+        self.run_thread_with_this(func=self.get_manager_metrics,
+                                  args=None)
+        self.total_timeout = os.getenv("TOTAL_TIMEOUT", DEFAULT_TOTAL_TIMEOUT)
+        self.execute_timeout = os.getenv("EXECUTE_TIMEOUT",
+                                         DEFAULT_EXECUTE_TIMEOUT)
+        self.max_threads = os.getenv("EXECUTE_TIMEOUT",
+                                     DEFAULT_EXECUTE_TIMEOUT)
+        start_time = time.time()
+        try:
+            while True:
+                while self.active_threads < self.max_threads:
+                    operation = randint(1, 3)
+                    if operation == 1:
+                        blueprint_name = \
+                            self.test_id+str(self.blueprints)
+                        self.logger.info(
+                            "uploading blueprint {0} with name {1}"
+                            .format(self.blueprints, blueprint_name))
+                        self.blueprints = self.blueprints + 1
+                        self.run_thread_with_this(
+                            func=self.upload_blueprint_retry,
+                            args=[blueprint_name])
+                    if operation == 2 and self.deployments < self.blueprints:
+                        self.logger.info("creating deployment {0}"
+                                         .format(self.deployments))
+                        blueprint_name = \
+                            str(self.test_id+str(self.deployments))
+                        deployment_name = blueprint_name
+                        params_dict = {"blueprint_id": blueprint_name,
+                                       "deployment_id": deployment_name}
+                        self.deployments = self.deployments + 1
+                        self.run_thread_with_this(
+                            func=self.create_deployment_retry,
+                            args=[params_dict])
+                    if operation == 3 and self.installed < self.deployments:
+                        self.logger.info("executing install on deployment {0}"
+                                         .format(self.installed))
+                        deployment_name = self.test_id+str(self.installed)
+                        self.installed = self.installed + 1
+                        self.run_thread_with_this(
+                            func=self.create_deployment_retry,
+                            args=[deployment_name])
+                if start_time - time.time() > self.execute_timeout:
+                    self._end_test()
```

Comment:
After calling this method the test loop will still run, you better check this with a short timeout.
Basically you should probably use this pattern:
try:
  while not timeout:
    ...
except Exception as e:
  ...
finally:
  self._end_test()

In addition it would be good to log the timeout before entering the while loop.